### PR TITLE
Team default channels gui

### PR DESCRIPTION
**Pull request by buoyad**

### CODE REVIEW

**Comment by buoyad** [comment]
cc @keybase/y2ksquad this is ready

File: shared/teams/common/channels-widget.tsx [hunk 1]
```diff
@@ -7,53 +7,66 @@ import useAutocompleter from './use-autocompleter'
...
-    channelnames,
-    onSelect,
-    filter
-  )
+  const {popup, popupAnchor, onKeyDown, setShowingPopup} = useAutocompleter(channelItems, onSelect, filter)
 
   return (
     <>
       <Kb.SearchFilter
+        // @ts-ignore complaining that popupAnchor is missing properties that SearchFilter has
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/teams/common/channels-widget.tsx]: TS requires all or none of the type annotations to be passed, I preferred to have it infer the type of `value` here.

**Comment by buoyad** [comment]
@jakob223 thanks, addressed feedback

**Comment by buoyad** [comment]
cc @joshblum I changed default channels RPCs to take team IDs

File: shared/teams/team/settings-tab/default-channels.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      [{teamID}],
+      result => {
+        setDefaultChannels([
+          {channelname: 'general', conversationIDKey: 'unused'},
+          ...(result.convs || []).map(conv => ({channelname: conv.channel, conversationIDKey: conv.convID})),
+        ])
+        setWaiting(false)
+      },
+      _ => {
+        // TODO what to do with error?
```
Comment (jakob223) [inline shared/teams/team/settings-tab/default-channels.tsx]: make a ticket to consider this, maybe?

**Pull request merged**